---
title: "Human Volunteer 16S Amplicon Analysis"
author: "Barry Hykes Jr. and Scott A. Handley"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: html_document
---
**RMarkdwon Description:** Additional analysis carried out as part of the published study, but excluded from inclusion in the results or materials and methods.

**Project Description:** Analysis of 16S rRNA amplicon data from a cohort of human volunteers (heatlhy men aged 18-35) randomized into three treatment groups: 1) control (no antibiotics), 2) narrow spectrum (oral vancomycin) or 3) broad-spectrum (oral vancomycin, ciprofloxacin, and metronidazole) antibiotics. Antibiotics were taken for 7-dats prior to vaccination with Rotarix (RVV), polysaccharide-pneumococcal (Pneumo 23) and tetanus-toxoid vaccine. The primary endpoint was difference in 28 days-post-vaccination anti-RV IgA. Secondary endpoints were proportion of volunteers with day 7 anti-RV IgA boosting (>=2 fold-increase), absolute and proportion of RV-antigen shedding, anti-RV, pneumococcal and anti-tetanus IgG.

**Primary Collaborator:**
Vanessa Harris (v.harris@aighd.org)

**Other relevant files:**
The following Phyloseq objects are available. Each is distinguished based on the 16S reference database used for taxonomic classification. RDP and Silva were processed through the species assignment workflow:

* ps0.human_volunteer.gg.RDS
* ps0.human_volunteer.silva.RDS
* ps0.human_volunteer.rdp.RDS

* Mapping file: mapping_human_volunteer.txt

**Workflow details:** The R commands below represent a full analysis of the following:

1) Sample quality control
2) ASV properties
3) Community Composition
4) Alpha diversity
5) Beta diversity
6) Differential abundance testing

```{r initiate-environment}
# Set default knitr option
knitr::opts_chunk$set(fig.width=8,
                      fig.height=6,
                      fig.path="./figures/",
                      dev='png',
                      warning=FALSE,
                      message=FALSE)
# Load libraries
library("tidyverse"); packageVersion("tidyverse")
library("reshape2"); packageVersion("reshape2")
library("plyr"); packageVersion("plyr")
library("phyloseq"); packageVersion("phyloseq")
library("RColorBrewer"); packageVersion("RColorBrewer")
library("vegan"); packageVersion("vegan")
library("gridExtra"); packageVersion("gridExtra")
library("knitr"); packageVersion("knitr")
library("plotly"); packageVersion("plotly")
library("microbiome"); packageVersion("microbiome")
library("ggpubr"); packageVersion("ggpubr")
library("data.table"); packageVersion("data.table")
library("pairwiseAdonis"); packageVersion("pairwiseAdonis")
library("DESeq2"); packageVersion("DESeq2")

# ggplot2 settings
# Set global theming
theme_set(theme_bw(base_size = 12))

# Set a random seed so that exact results can be reproduced
set.seed(1000)

```
##Read in data

```{r initiate-data}
# Load Phyloseq Object
# Selected RDP due to it's up-to-date nature and conservative taxonomy. Other files are also valid for anlysis but are not explored here but are available in the ./data/ directory
ps0 <- readRDS("~/Dropbox/Research/human_volunteer/data/ps0.human_volunteer.rdp.RDS")
ps0

# Load mapping file
map <- import_qiime_sample_data("~/Dropbox/Research/human_volunteer/data/mapping_human_volunteer.txt")
ps0 <- merge_phyloseq(ps0, map)
ps0

# View sample variables & generate basic stats
sample_variables(ps0)
sd(sample_sums(ps0))
get_taxa_unique(ps0, "Phylum")
ntaxa(ps0)

```
##Factor reordering and renaming

```{r factor-adjustments}
# Order and update randomization arms factor
sample_data(ps0)$randomization_arm
sample_data(ps0)$randomization_arm <- factor(sample_data(ps0)$randomization_arm, levels = c("No_antibiotics", "Narrow_spectrum_antibiotics", "Broad_spectrum_antibiotics"))
sample_data(ps0)$randomization_arm
sample_data(ps0)$randomization_arm <- factor(sample_data(ps0)$randomization_arm, labels = c("No antibiotics", "Narrow spectrum antibiotics", "Broad spectrum antibiotics"))
sample_data(ps0)$randomization_arm

```
##Metadata analysis

```{r metadata-analysis}
# Smaple data overview
sample_variables(ps0)

# Prepare a simple data frame of sample data
sd <- ps0 %>%
  sample_data() %>%
  as.data.frame() %>%
  as.tibble()

# Age by randomization arm
meta.p.age <- ggplot(sd, aes(x = Age)) +
  geom_histogram(binwidth = 5) +
  facet_grid(~randomization_arm) +
  labs(title = "Age by Randomization Arm", y = "Count", x = "Age (years)")
plot_grid(meta.p.age, labels = "A")

# Height and Weight by randomization arm
meta.p.height <- ggplot(sd, aes(x = Length)) +
  geom_histogram(binwidth = 5) +
  labs(y = "Count", x = "Height (cm)", caption = "Dashed line = Median Value") +
  facet_wrap(~randomization_arm, ncol = 1) +
  geom_vline(aes(xintercept=median(Length)), lty = 2, lwd = 0.2)

meta.p.weight <- ggplot(sd, aes(x = Weight)) +
  geom_histogram(binwidth = 5) +
  labs(y = "Count", x = "Weight (kg)", caption = "Dashed line = Median Value") +
  facet_wrap(~randomization_arm, ncol = 1) +
  geom_vline(aes(xintercept=median(Weight)), lty = 2, lwd = 0.2)
plot_grid(meta.p.height, meta.p.weight, labels = c("A", "B"), ncol = 2)

# Rotavirus Ag shedding by randomization arm
sd.1 <- sd %>%
  gather(RV_Ag_shedding_d0, RV_Ag_shedding_d1, RV_Ag_shedding_d2, RV_Ag_shedding_d3, RV_Ag_shedding_d4, RV_Ag_shedding_d5, RV_Ag_shedding_d6, key = "RV_Ag_shedding_day", value = "Shed_RV_Ag")
levels(as.factor(sd.1$RV_Ag_shedding_day))

meta.p.rv_ag.shedding <- ggplot(sd.1, aes(x = RV_Ag_shedding_day, y = log10(Shed_RV_Ag), color = randomization_arm)) +
  geom_point(size = 3) +
  stat_summary(aes(group = randomization_arm, color = randomization_arm), fun.y=median, geom="line", lwd = 0.5, lty = 2) +
  labs(color = "Randomization Arm", y = "Shed RV Antigen (log10)") +
  labs(x = "") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
plot_grid(meta.p.rv_ag.shedding, labels = "A")

# Rotavirus serum IgA
sd.2 <- sd %>%
  gather(d7_RV_IgA, d14_RV_IgA, d28_RV_IgA, key = "RV_IgA_serum_day", value = "RV_IgA")
levels(as.factor(sd.2$RV_IgA_serum_day))
sd.2$RV_IgA_serum_day <- factor(sd.2$RV_IgA_serum_day, levels = c("d7_RV_IgA", "d14_RV_IgA", "d28_RV_IgA"))
levels(as.factor(sd.2$RV_IgA_serum_day))

meta.p.rv_serum_IgA <- ggplot(sd.2, aes(x = RV_IgA_serum_day, y = log10(RV_IgA), color = randomization_arm, size = log10(pre_RV_IgA))) +
  geom_point() +
  stat_summary(aes(group = randomization_arm, color = randomization_arm), fun.y=median, geom="line", lwd = 0.5, lty = 2) +
  labs(color = "Randomization Arm", y = "Rotavirus IgA (log10)") +
  labs(x = "") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
plot_grid(meta.p.rv_serum_IgA, labels = "A")

ggplot(sd.2, aes(x = RV_IgA_serum_day, y = RV_IgA)) +
  geom_point()

# Pneumococcus IgG
sd.3 <- sd %>%
  gather(pneumo_IgG_pre, pneumo_IgG_7, pneumo_IgG_14, pneumo_IgG_28, key = "Pneumo_IgG_day", value = "Pneumo_IgG")
levels(as.factor(sd.3$Pneumo_IgG_day))
sd.3$Pneumo_IgG_day <- factor(sd.3$Pneumo_IgG_day, levels = c("pneumo_IgG_pre", "pneumo_IgG_7", "pneumo_IgG_14", "pneumo_IgG_28"))

meta.p.pneumo_IgG <- ggplot(sd.3, aes(x = Pneumo_IgG_day, y = log10(Pneumo_IgG), color = randomization_arm)) +
  geom_point(size = 3) +
  stat_summary(aes(group = randomization_arm, color = randomization_arm), fun.y=median, geom="line", lwd = 0.5, lty = 2) +
  labs(color = "Randomization Arm", y = "Pneumococcus IgG (log10)") +
  labs(x = "") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
plot_grid(meta.p.pneumo_IgG, labels = "A")

# Tetanus IfF
sd.4 <- sd %>%
  gather("tetanus_IgG_pre", "tetanus_IgG_d7", "tetanus_IgG_d14", "tetanus_IgG_d28", key = "Tetanus_IgG_day", value = "Tetanus_IgG")
levels(as.factor(sd.4$Tetanus_IgG_day))
sd.4$Tetanus_IgG_day <- factor(sd.4$Tetanus_IgG_day, levels = c("tetanus_IgG_pre", "tetanus_IgG_d7", "tetanus_IgG_d14", "tetanus_IgG_d28"))
levels(as.factor(sd.4$Tetanus_IgG_day))

meta.p.tetanus_IgG <- ggplot(sd.4, aes(x = Tetanus_IgG_day, y = log10(Tetanus_IgG), color = randomization_arm)) +
  geom_point(size = 3) +
  stat_summary(aes(group = randomization_arm, color = randomization_arm), fun.y=median, geom="line", lwd = 0.5, lty = 2) +
  labs(color = "Randomization Arm", y = "Tetanus IgG (log10)") +
  labs(x = "") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
plot_grid(meta.p.tetanus_IgG, labels = "A")

plot_grid(meta.p.rv_serum_IgA, meta.p.pneumo_IgG, meta.p.tetanus_IgG, meta.p.rv_ag.shedding, labels = c("A", "B", "C", "D"), nrow = 2, ncol = 2 )

# Pre Rotabirus IgA
summary(sd$pre_RV_IgA)

ggplot(sd, aes(x = pre_RV_IgA)) +
  geom_histogram(binwidth = 250) +
  labs(y = "Count") +
  facet_wrap(~randomization_arm)

ggplot(sd, aes(x = randomization_arm, y = log10(pre_RV_IgA))) +
  geom_boxplot() +
  geom_point(aes(color = d7_rota_boost_updated), size = 3, shape = 1) +
  labs(x = "")

fit <- aov(log10(pre_RV_IgA) ~ randomization_arm, data=sd)
summary(fit)
TukeyHSD(fit)

```
##ASV summary statistics

```{r data-assessment}
# Create a new data frame of the sorted row sums, a column of sorted values from 1 to the total number of individuals/counts for each RSV and a categorical variable stating these are all RSVs.
readsumsdf = data.frame(nreads = sort(taxa_sums(ps0), TRUE), 
                        sorted = 1:ntaxa(ps0),
                        type = "ASVs")


# Add a column of sample sums (total number of individuals per sample)
readsumsdf = rbind(readsumsdf,
                   data.frame(nreads = sort(sample_sums(ps0), TRUE),
                              sorted = 1:nsamples(ps0),
                              type = "Samples"))

# Make a data frame with a column for the read counts of each sample for histogram production
sample_sum_df <- data.frame(sum = sample_sums(ps0))

# Make plots
# Generates a bar plot with # of reads (y-axis) for each taxa. Sorted from most to least abundant
# Generates a second bar plot with # of reads (y-axis) per sample. Sorted from most to least
p.reads = ggplot(readsumsdf, aes(x = sorted, y = nreads)) +
  geom_bar(stat = "identity") +
  ggtitle("ASV Assessment") +
  scale_y_log10() +
  facet_wrap(~type, scales = "free") +
  ylab("# of Reads")

# Histogram of the number of Samples (y-axis) at various read depths
p.reads.hist <- ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "firebrick3", binwidth = 150) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Counts") +
  ylab("# of Samples")

# Final plot, side-by-side
grid.arrange(p.reads, p.reads.hist, ncol = 2)

# Basic summary statistics
summary(sample_sums(ps0))

```

##Prevalence assessment

```{r prevalenca-assessment}
# Intermediate data subsetting for prevelance plotting
# Full subsetting with removal of zero count data will be performed in a subsequent chunk
ps1.d_9 <- subset_samples(ps1, day == "-9")
ps1.d0 <- subset_samples(ps1, day == "0")
ps1.d7 <- subset_samples(ps1, day == "7")

# Prevelance estimation
# Calculate feature prevelance across the data set
prevdf.d_9 <- apply(X = otu_table(ps1.d_9),MARGIN = ifelse(taxa_are_rows(ps1.d_9), yes = 1, no = 2),FUN = function(x){sum(x > 0)})
prevdf.d0 <- apply(X = otu_table(ps1.d0),MARGIN = ifelse(taxa_are_rows(ps1.d0), yes = 1, no = 2),FUN = function(x){sum(x > 0)})
prevdf.d7 <- apply(X = otu_table(ps1.d7),MARGIN = ifelse(taxa_are_rows(ps1.d7), yes = 1, no = 2),FUN = function(x){sum(x > 0)})

# Add taxonomy and total read counts to prevdf
prevdf.d_9 <- data.frame(Prevalence = prevdf.d_9, TotalAbundance = taxa_sums(ps1.d_9), tax_table(ps1.d_9))
prevdf.d0 <- data.frame(Prevalence = prevdf.d0, TotalAbundance = taxa_sums(ps1.d0), tax_table(ps1.d0))
prevdf.d7 <- data.frame(Prevalence = prevdf.d7, TotalAbundance = taxa_sums(ps1.d7), tax_table(ps1.d7))

# Create a table of Phylum, their mean abundances across all samples, and the number of samples they were detected in
plyr::ddply(prevdf.d_9, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})
plyr::ddply(prevdf.d0, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})
plyr::ddply(prevdf.d7, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})

# Reorder based on visual inspection of relative dominance of each taxa for plotting
prevdf.d_9$Phylum <- factor(prevdf.d_9$Phylum, levels = c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Verrucomicrobia", "Lentisphaerae", "Fusobacteria", "Elusimicrobia", "Spirochaetes", "Synergistetes", "Tenericutes"))
prevdf.d0$Phylum <- factor(prevdf.d0$Phylum, levels = c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Verrucomicrobia", "Lentisphaerae", "Fusobacteria", "Elusimicrobia", "Spirochaetes", "Synergistetes", "Tenericutes"))
prevdf.d7$Phylum <- factor(prevdf.d7$Phylum, levels = c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Verrucomicrobia", "Lentisphaerae", "Fusobacteria", "Elusimicrobia", "Spirochaetes", "Synergistetes", "Tenericutes"))

#Prevalence plot - t1
prevdf1.d_9 <- subset(prevdf.d_9, Phylum %in% get_taxa_unique(ps1.d_9, "Phylum"))
p.prevdf1.d_9 <- ggplot(prevdf1.d_9, aes(TotalAbundance, Prevalence / nsamples(ps1.d_9),color=Family)) +
  # geom_hline(yintercept = 0.03, alpha = 0.5, linetype = 2) +
  geom_point(size = 1.5, alpha = 0.7) +
  scale_x_log10(limits = c(1,10000), breaks = c(100,10000)) +
  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_grid(~Phylum) +
  theme(legend.position="None") +
  theme(axis.title.x = element_blank()) +
  ggtitle("Day -9")

#Prevalence plot - t2
prevdf1.d0 <- subset(prevdf.d0, Phylum %in% get_taxa_unique(ps1.d0, "Phylum"))
p.prevdf1.d0 <- ggplot(prevdf1.d0, aes(TotalAbundance, Prevalence / nsamples(ps1.d0),color=Family)) +
  # geom_hline(yintercept = 0.03, alpha = 0.5, linetype = 2) +
  geom_point(size = 1.5, alpha = 0.7) +
  scale_x_log10(limits = c(1,10000), breaks = c(100,10000)) +
  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_grid(~Phylum) +
  theme(legend.position="None") +
  theme(axis.title.x = element_blank()) +
  ggtitle("Day 0")

#Prevalence plot - t3
prevdf1.d7 <- subset(prevdf.d7, Phylum %in% get_taxa_unique(ps1.d7, "Phylum"))
p.prevdf1.d7 <- ggplot(prevdf1.d7, aes(TotalAbundance, Prevalence / nsamples(ps1.d7),color=Family)) +
  # geom_hline(yintercept = 0.03, alpha = 0.5, linetype = 2) +
  geom_point(size = 1.5, alpha = 0.7) +
  scale_x_log10(limits = c(1,10000), breaks = c(100,10000)) +
  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_grid(~Phylum) +
  theme(legend.position="None") +
  theme(axis.title.x = element_blank()) +
  ggtitle("Day 7")

grid.arrange(p.prevdf1.d_9, p.prevdf1.d0, p.prevdf1.d7, nrow = 3)

```

```{r LRT-diff-abund-testing-data-prep, include=FALSE}
# Time series should be set as a factor per DeSeq2 recommendations
class(sample_data(ps1.con)$day)
sample_data(ps1.con)$day <- factor(sample_data(ps1.con)$day)
class(sample_data(ps1.con)$day)

class(sample_data(ps1.narrow)$day)
sample_data(ps1.narrow)$day <- factor(sample_data(ps1.narrow)$day)
class(sample_data(ps1.narrow)$day)

class(sample_data(ps1.broad)$day)
sample_data(ps1.broad)$day <- factor(sample_data(ps1.broad)$day)
class(sample_data(ps1.broad)$day)

# Calculate geometric means prior to estimate size factors
# This is required for data sets with lots of zeros
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}

```
#**Likelihood ratio test:** Identifying taxa differentially abundant in one or more time points between 1) patients who boosted versus thoses that did not and 2) patients that shed rotavirus in their stool and those that did not

##1) Rotavirus Boost: Narrow Spectrum Antibiotics

```{r diff-abund-rotaboost-narrow-LRT}
# Test for taxa which at one or more time points after time 0 showed a d7-rota-boost-specific effect
# Convert phyloseq object to DeSeq2 table
ds.boost.narrow.LRT <- phyloseq_to_deseq2(ps1.narrow, ~d7_rota_boost_updated + day + d7_rota_boost_updated:day)

# Calculate geometric means prior to estimate size factors
geoMeans.ds.boost.narrow.LRT <- apply(counts(ds.boost.narrow.LRT), 1, gm_mean)
ds.boost.narrow.LRT <- estimateSizeFactors(ds.boost.narrow.LRT, geoMeans = geoMeans.ds.boost.narrow.LRT)

# Run DeSeq2
dds.boost.narrow.LRT <- DESeq(ds.boost.narrow.LRT, test="LRT", reduced = ~d7_rota_boost_updated + day)

# Tabulate results
res.dds.boost.narrow.LRT = results(dds.boost.narrow.LRT)
res.dds.boost.narrow.LRT$symbol <- mcols(dds.boost.narrow.LRT)$symbol
head(res.dds.boost.narrow.LRT[order(res.dds.boost.narrow.LRT$padj), ], 4)
write.table(res.dds.boost.narrow.LRT, file = "./results/deseq_rotaboost_narrow_LRT.txt", sep = "\t")
mcols(res.dds.boost.narrow.LRT)
summary(res.dds.boost.narrow.LRT)

# Plot taxa with lowest p-value
fiss.dds.boost.narrow.LRT <- plotCounts(dds.boost.narrow.LRT, which.min(res.dds.boost.narrow.LRT$padj), 
                   intgroup = c("day","d7_rota_boost_updated"), returnData = TRUE)
ggplot(fiss.dds.boost.narrow.LRT,
  aes(x = as.numeric(day), y = count, color = d7_rota_boost_updated, group = d7_rota_boost_updated)) + 
  geom_point() + geom_smooth(se = FALSE, method = "loess") + scale_y_log10()

```
##2) Shedding: Narrow Spectrum Antibiotics Only

```{r diff-abund-Shedding-narrow-LRT}
# Test for taxa which at one or more time points after time 0 showed a Shedding-specific effect
# Convert phyloseq object to DeSeq2 table
ds.Shedding.narrow.LRT <- phyloseq_to_deseq2(ps1.narrow, ~Shedding + day + Shedding:day)

# Calculate geometric means prior to estimate size factors
geoMeans.ds.Shedding.narrow.LRT <- apply(counts(ds.Shedding.narrow.LRT), 1, gm_mean)
ds.Shedding.narrow.LRT <- estimateSizeFactors(ds.Shedding.narrow.LRT, geoMeans = geoMeans.ds.Shedding.narrow.LRT)

# Run DeSeq2
dds.Shedding.narrow.LRT <- DESeq(ds.Shedding.narrow.LRT, test="LRT", reduced = ~Shedding + day)

# Tabulate results
res.dds.Shedding.narrow.LRT = results(dds.Shedding.narrow.LRT)
res.dds.Shedding.narrow.LRT$symbol <- mcols(dds.Shedding.narrow.LRT)$symbol
head(res.dds.Shedding.narrow.LRT[order(res.dds.Shedding.narrow.LRT$padj), ], 4)
write.table(res.dds.Shedding.narrow.LRT, file = "./results/deseq_Shedding_narrow_LRT.txt", sep = "\t")
mcols(res.dds.Shedding.narrow.LRT)
summary(res.dds.Shedding.narrow.LRT)

# Plot taxa with lowest p-value
fiss.dds.Shedding.narrow.LRT <- plotCounts(dds.Shedding.narrow.LRT, which.min(res.dds.Shedding.narrow.LRT$padj), 
                   intgroup = c("day","d7_rota_boost_updated"), returnData = TRUE)
ggplot(fiss.dds.Shedding.narrow.LRT,
  aes(x = as.numeric(day), y = count, color = d7_rota_boost_updated, group = d7_rota_boost_updated)) + 
  geom_point() + geom_smooth(se = FALSE, method = "loess") + scale_y_log10()

```

```{r compare-assignments}
library(daff)
diff.rdp_gg <- diff_data(df.all.rdp, df.all.gg)
write_diff(diff.rdp_gg, file = "./results/diff_rdp_gg.csv")
render_diff(diff.rdp_gg)

diff.rdp_silva <- diff_data(df.all.rdp, df.all.silva)
write_diff(diff.rdp_silva, file = "./results/diff.rdp_silva.csv")
render_diff(diff.rdp_silva)

diff.gg_silva <- diff_data(df.all.gg, df.all.silva)
write_diff(diff.gg_silva, file = "./results/diff.gg_silva.csv")
render_diff(diff.gg_silva)

```