---
title: "Human Volunteer 16S Amplicon Analysis"
author: "Scott A. Handley"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: html_document
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=8,
                      fig.height=6,
                      fig.path="./figures/",
                      warning=FALSE,
                      message=FALSE)
```
##Initiate environment

```{r initiate-environment}
library("tidyverse")
packageVersion("tidyverse")
library("phyloseq")
packageVersion("phyloseq")
library("RColorBrewer")
packageVersion("RColorBrewer")
library("vegan")
packageVersion("vegan")
library("gridExtra")
packageVersion("gridExtra")
library("knitr")
packageVersion("knitr")
library("DESeq2")
packageVersion("DeSeq2")
library("plotly")
packageVersion("plotly")
library("microbiome")
packageVersion("microbiome")
library("ggpubr")
packageVersion("ggpubr")
library("randomForest")
packageVersion("randomForest")
library("data.table")
packageVersion("data.table")
library("HH")
packageVersion("HH")

```

```{r global-theme-settings, include=FALSE}
# Set global theming
theme_set(theme_bw(base_size = 10,
                   base_family = "Helvetica"))

```
##Load phyloseq data generated from dada2

```{r initiate-data}
# Load Phyloseq object generated in HIV_oral_preprocessing.Rmd
# Options include GreenGenes, SILVA, RDP and HitDB
ps0 <- readRDS("~/Dropbox/Research/human_volunteer/data/ps0.Harris.rdp.RDS")
ps0

# Load mapping file
map <- import_qiime_sample_data("~/Dropbox/Research/human_volunteer/data/human_volunteer_mapping_USE.txt")
ps0 <- merge_phyloseq(ps0, map)
ps0

# View sample variables & generate basic stats
sample_variables(ps0)
sd(sample_sums(ps0))
get_taxa_unique(ps0, "Phylum")

```
##Factor reordering and renaming

```{r factor-adjustments}
# Order randomization arms 
sample_data(ps0)$randomization_arm
sample_data(ps0)$randomization_arm <- factor(sample_data(ps0)$randomization_arm, levels = c("No_antibiotics", "Narrow_spectrum_antibiotics", "Broad_spectrum_antibiotics"))
sample_data(ps0)$randomization_arm
sample_data(ps0)$randomization_arm <- factor(sample_data(ps0)$randomization_arm, labels = c("No antibiotics", "Narrow spectrum antibiotics", "Broad spectrum antibiotics"))
sample_data(ps0)$randomization_arm

# Order sample days (Pre -> zero -> 7)
sample_data(ps0)$day
sample_data(ps0)$day <- factor(sample_data(ps0)$day, levels = c("Pre", "0", "7"))
sample_data(ps0)$day

```
##ASV summary statistics

```{r data-assessment}
# Create a new data frame of the sorted row sums, a column of sorted values from 1 to the total number of individuals/counts for each RSV and a categorical variable stating these are all RSVs.
readsumsdf = data.frame(nreads = sort(taxa_sums(ps0), TRUE), 
                        sorted = 1:ntaxa(ps0),
                        type = "RSVs")


# Add a column of sample sums (total number of individuals per sample)
readsumsdf = rbind(readsumsdf,
                   data.frame(nreads = sort(sample_sums(ps0), TRUE),
                              sorted = 1:nsamples(ps0),
                              type = "Samples"))

# Make a data frame with a column for the read counts of each sample for histogram production
sample_sum_df <- data.frame(sum = sample_sums(ps0))

# Make plots
# Generates a bar plot with # of reads (y-axis) for each taxa. Sorted from most to least abundant
# Generates a second bar plot with # of reads (y-axis) per sample. Sorted from most to least
p.reads = ggplot(readsumsdf, aes(x = sorted, y = nreads)) +
  geom_bar(stat = "identity") +
  ggtitle("RSV Assessment") +
  scale_y_log10() +
  facet_wrap(~type, scales = "free") +
  ylab("# of Reads")

# Histogram of the number of Samples (y-axis) at various read depths
p.reads.hist <- ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "firebrick3", binwidth = 150) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  ylab("# of Samples")

# Final plot, side-by-side
grid.arrange(p.reads, p.reads.hist, ncol = 2)

# Basic summary statistics
summary(sample_sums(ps0))

```
There appear to be quite a few samples with low numbers of ASVs. Examination in next code chunk will reveal a threshold for removing samples.

##Bad sample identification and removal

```{r sample-removal-identification}
# Format a data table to combine sample summary data with sample variable data
ss <- sample_sums(ps0)
sd <- as.data.frame(sample_data(ps0)) # useful to coerce the phyloseq object into an R data frame
ss.df <- merge(sd, data.frame("RSVs" = ss), by ="row.names") # merge ss with sd by row names. Rename ss to RSVs in the new data frame

# Plot the data by the treatment variable

y = 100 # Set a threshold for the minimum number of acceptable reads. Can start as a guess
x = "randomization_arm" # Set the x-axis variable you want to examine
label = "Description" # This is the label you want to overlay on the points that are below threshold y. Should be something sample specific

# Plot
p.ss.boxplot<- ggplot(ss.df, aes_string(x, y = "RSVs")) + # x is what you assigned it above
  geom_boxplot(outlier.colour="NA", aes(group = randomization_arm)) +
  scale_y_log10() +
  geom_hline(yintercept = y, lty = 2) + # Draws a dashed line across the threshold you set above as y
  geom_jitter(alpha = 0.5, width = 0.15, size = 3) +
  #geom_text(data = subset(ss.df, RSVs <= y), aes_string(x, y="RSVs", label=label), size=2, hjust -1) +# This labels a subset that fall below threshold variable y and labels them with the label variable
  ggtitle("Number of ASV") +
  ylab("ASV (log10)") +
  facet_grid(~day) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  theme(legend.position = "NULL")
p.ss.boxplot

# Remove samples with fewer than 100 sequences
# Save samples that fell below the selected threshold
ps.failed <- prune_samples(sample_sums(ps0)<y, ps0)
ps.failed
failed.samples <- data.frame(sample_names(ps.failed), sample_data(ps.failed)$sample_ID, sample_data(ps.failed)$tube_ID)
write.table(failed.samples, file = "./results/failed_sampels.txt", sep = "\t")

ps0 <- prune_samples(sample_sums(ps0)>=y, ps0)
ps0
min(sample_sums(ps0))

```
There does not appear to be a relationship between low numbers of ASVs, treatment or time-point. So removing samples that behave unexpectedly (< 100 sequences).

##Overall sample relationship to evaluate sample outliers

```{r outlier-sample-evaluation}
# Outlier evaluation
out.bray <- ordinate(ps0, method = "MDS", distance = "bray")
p.MDS.outlier <- plot_ordination(ps0, out.bray, color = "randomization_arm", axes = c(1,2)) +
  theme_bw() +
  facet_grid(~day) +
  geom_point(size = 2) +
  ggtitle("MDS of Bray Distances \nOutlier Evaluation") +
  stat_ellipse(type = "norm", geom = "polygon", alpha = 1/10, aes(fill = randomization_arm))
p.MDS.outlier

```
There is no distiguishable feature to identify samples as obvious outliers using this projection. No samples will be removed based on MDS of Bray-curtis distances.

##Prevalence assessment and taxon filtering

```{r prevalence-assessment}
# Begin by removing sequences that were not classified as Bacteria or were classified as either mitochondria or chlorplast

ps0 # Check the number of taxa prior to removal
ps1 <- ps0 %>%
  subset_taxa(
    Kingdom == "Bacteria" &
    Family  != "mitochondria" &
    Class   != "Chloroplast"
  )
ps1 # Confirm that the taxa were removed

# Intermediate data subsetting for prevelance plotting
# Full subsetting with removal of zero count data will be performed in a subsequent chunk
ps1.d_9 <- subset_samples(ps1, day == "Pre")
ps1.d0 <- subset_samples(ps1, day == "0")
ps1.d7 <- subset_samples(ps1, day == "7")

# Prevelance estimation
# Calculate feature prevelance across the data set
prevdf.d_9 <- apply(X = otu_table(ps1.d_9),MARGIN = ifelse(taxa_are_rows(ps1.d_9), yes = 1, no = 2),FUN = function(x){sum(x > 0)})
prevdf.d0 <- apply(X = otu_table(ps1.d0),MARGIN = ifelse(taxa_are_rows(ps1.d0), yes = 1, no = 2),FUN = function(x){sum(x > 0)})
prevdf.d7 <- apply(X = otu_table(ps1.d7),MARGIN = ifelse(taxa_are_rows(ps1.d7), yes = 1, no = 2),FUN = function(x){sum(x > 0)})

# Add taxonomy and total read counts to prevdf
prevdf.d_9 <- data.frame(Prevalence = prevdf.d_9, TotalAbundance = taxa_sums(ps1.d_9), tax_table(ps1.d_9))
prevdf.d0 <- data.frame(Prevalence = prevdf.d0, TotalAbundance = taxa_sums(ps1.d0), tax_table(ps1.d0))
prevdf.d7 <- data.frame(Prevalence = prevdf.d7, TotalAbundance = taxa_sums(ps1.d7), tax_table(ps1.d7))

# Create a table of Phylum, their mean abundances across all samples, and the number of samples they were detected in
plyr::ddply(prevdf.d_9, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})
plyr::ddply(prevdf.d0, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})
plyr::ddply(prevdf.d7, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})

# Reorder based on visual inspection of relative dominance of each taxa for plotting
prevdf.d_9$Phylum <- factor(prevdf.d_9$Phylum, levels = c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Verrucomicrobia", "Lentisphaerae", "Fusobacteria", "Elusimicrobia", "Spirochaetes", "Synergistetes", "Tenericutes"))
prevdf.d0$Phylum <- factor(prevdf.d0$Phylum, levels = c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Verrucomicrobia", "Lentisphaerae", "Fusobacteria", "Elusimicrobia", "Spirochaetes", "Synergistetes", "Tenericutes"))
prevdf.d7$Phylum <- factor(prevdf.d7$Phylum, levels = c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Verrucomicrobia", "Lentisphaerae", "Fusobacteria", "Elusimicrobia", "Spirochaetes", "Synergistetes", "Tenericutes"))

#Prevalence plot - t1
prevdf1.d_9 <- subset(prevdf.d_9, Phylum %in% get_taxa_unique(ps1.d_9, "Phylum"))
p.prevdf1.d_9 <- ggplot(prevdf1.d_9, aes(TotalAbundance, Prevalence / nsamples(ps1.d_9),color=Family)) +
  # geom_hline(yintercept = 0.03, alpha = 0.5, linetype = 2) +
  geom_point(size = 1.5, alpha = 0.7) +
  scale_x_log10(limits = c(1,10000), breaks = c(100,10000)) +
  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_grid(~Phylum) +
  theme(legend.position="None") +
  theme(axis.title.x = element_blank()) +
  ggtitle("Day -9")

#Prevalence plot - t2
prevdf1.d0 <- subset(prevdf.d0, Phylum %in% get_taxa_unique(ps1.d0, "Phylum"))
p.prevdf1.d0 <- ggplot(prevdf1.d0, aes(TotalAbundance, Prevalence / nsamples(ps1.d0),color=Family)) +
  # geom_hline(yintercept = 0.03, alpha = 0.5, linetype = 2) +
  geom_point(size = 1.5, alpha = 0.7) +
  scale_x_log10(limits = c(1,10000), breaks = c(100,10000)) +
  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_grid(~Phylum) +
  theme(legend.position="None") +
  theme(axis.title.x = element_blank()) +
  ggtitle("Day 0")

#Prevalence plot - t3
prevdf1.d7 <- subset(prevdf.d7, Phylum %in% get_taxa_unique(ps1.d7, "Phylum"))
p.prevdf1.d7 <- ggplot(prevdf1.d7, aes(TotalAbundance, Prevalence / nsamples(ps1.d7),color=Family)) +
  # geom_hline(yintercept = 0.03, alpha = 0.5, linetype = 2) +
  geom_point(size = 1.5, alpha = 0.7) +
  scale_x_log10(limits = c(1,10000), breaks = c(100,10000)) +
  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_grid(~Phylum) +
  theme(legend.position="None") +
  theme(axis.title.x = element_blank()) +
  ggtitle("Day 7")

grid.arrange(p.prevdf1.d_9, p.prevdf1.d0, p.prevdf1.d7, nrow = 3)

```
There are a number of low prevalent bacteria. There is a notable reduction in both Firmicutes and Bacteroidetes following antibiotic treatment. This is not subset by treatment (randomization_arm), but the overall reduction suggests the antibiotics reduced the prevalence of many taxa following treatment. There is a visible restoration by day 7.

##Common data transformations

```{r data-transform}
# Transform to Realative abundances
ps1.ra <- transform_sample_counts(ps1, function(OTU) OTU/sum(OTU))

# Transform to Proportional Abundance
ps1.prop <- transform_sample_counts(ps1, function(x) min(sample_sums(ps1)) * x/sum(x))

# Log transformation moves to a more normal distribution
ps1.log <- transform_sample_counts(ps1, function(x) log(1 + x))

# View how each function altered count data
par(mfrow=c(1,4))
plot(sort(sample_sums(ps1), TRUE), type = "o", main = "Native", ylab = "RSVs", xlab = "Samples")
plot(sort(sample_sums(ps1.log), TRUE), type = "o", main = "log Transfromed", ylab = "RSVs", xlab = "Samples")
plot(sort(sample_sums(ps1.ra), TRUE), type = "o", main = "Relative Abundance", ylab = "RSVs", xlab = "Samples")
plot(sort(sample_sums(ps1.prop), TRUE), type = "o", main = "Proportional Abundance", ylab = "RSVs", xlab = "Samples")
par(mfrow=c(1,1))

# Histograms of the non-transformed data vs. the transformed data can address the shift to normality
p.nolog <- qplot(rowSums(otu_table(ps1))) + ggtitle("Raw Counts") +
  theme_bw() +
  xlab("Row Sum") +
  ylab("# of Samples")

p.log <- qplot(log10(rowSums(otu_table(ps1)))) +
  ggtitle("log10 transformed counts") +
  theme_bw() +
  xlab("Row Sum") +
  ylab("# of Samples")

grid.arrange(p.nolog, p.log, ncol = 2)

```
Log10 transformation does tend to shift the data to normality, but it is not striking and may 'overshift' the data. Should consider using log10 transformed data for future analysis, but consider other transformations as needed.

##Subsetting

```{r subsetting}
# Subset. Always subset after taxa/sample filtering
# Will go ahead and log transform in preparation for testing the effects on analysis

# Time 1 (Day -9)
ps1.t1 <- subset_samples(ps1, day == "Pre")
ps1.t1
ps1.t1 <- prune_samples(sample_sums(ps1.t1) > 0, ps1.t1)
ps1.t1
ps1.t1.log <- transform_sample_counts(ps1.t1, function(x) log(1 + x))

# Time 2 (Day 0)
ps1.t2 <- subset_samples(ps1, day == "0")
ps1.t2
ps1.t2 <- prune_samples(sample_sums(ps1.t2) > 0, ps1.t2)
ps1.t2
ps1.t2.log <- transform_sample_counts(ps1.t2, function(x) log(1 + x))

# Time 3 (Day 7)
ps1.t3 <- subset_samples(ps1, day == "7")
ps1.t3
ps1.t3 <- prune_samples(sample_sums(ps1.t3) > 0, ps1.t3)
ps1.t3
ps1.t3.log <- transform_sample_counts(ps1.t3, function(x) log(1 + x))

# No antibiotics (control)
ps1.con <- subset_samples(ps1, randomization_arm == "No antibiotics")
ps1.con
ps1.con <- prune_samples(sample_sums(ps1.con) > 0, ps1.con)
ps1.con
ps1.con.log <- transform_sample_counts(ps1.con, function(x) log(1 + x))

# Narrow spectrum antibiotics
ps1.narrow <- subset_samples(ps1, randomization_arm == "Narrow spectrum antibiotics")
ps1.narrow
ps1.narrow <- prune_samples(sample_sums(ps1.narrow) > 0, ps1.narrow)
ps1.narrow
ps1.narrow.log <- transform_sample_counts(ps1.narrow, function(x) log(1 + x))

# Broad spectrum antibiotics
ps1.broad <- subset_samples(ps1, randomization_arm == "Broad spectrum antibiotics")
ps1.broad
ps1.broad <- prune_samples(sample_sums(ps1.broad) > 0, ps1.broad)
ps1.broad
ps1.broad.log <- transform_sample_counts(ps1.broad, function(x) log(1 + x))

```
##Community composition plotting

```{r community-composition-plots}
# Community composition - Baseline and Week 24 timepoints combined
# Phyla level plots
# Melt to long format (for ggploting) 
# Prune out phyla below % in each sample

threshold = 0.03

# Time 1 (Day -9)
ps1_phylum.t1 <- ps1.t1 %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > threshold) %>%                    # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

# Convert Sample No to a factor because R is weird sometime
ps1_phylum.t1$patient_ID <- as.factor(ps1_phylum.t1$patient_ID)

# Creat a Phyla level table for external analysis
# write.table(ps1_phylum, file = "./results/phylum.txt", sep="\t")

# Plot - Phylum - Time 1 (Day -9)
p.comm.bar.t1 <- ggplot(ps1_phylum.t1, aes(x = patient_ID, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity") +
  facet_wrap(~randomization_arm, scales = "free_x") +
  ggtitle("Day -9\nAbundant Phylum (> 3%) in All Samples") +
  ylab("Relative Abundance") +
  theme(legend.position = "bottom") +
  scale_fill_manual(values = c('#1b9e77','#d95f02','#7570b3','#e7298a','#66a61e'))
p.comm.bar.t1

# Generate interavtive bar plot
ggplotly(p.comm.bar.t1)

# Time 2 (Day 0)
ps1_phylum.t2 <- ps1.t2 %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > threshold) %>%                    # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

# Convert Sample No to a factor because R is weird sometime
ps1_phylum.t2$patient_ID <- as.factor(ps1_phylum.t2$patient_ID)

# Creat a Phyla level table for external analysis
# write.table(ps1_phylum, file = "./results/phylum.txt", sep="\t")

# Plot - Phylum - Time 2 (Day 0)
p.comm.bar.t2 <- ggplot(ps1_phylum.t2, aes(x = patient_ID, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity") +
  facet_wrap(~randomization_arm, scales = "free_x") +
  ggtitle("Day 0\nAbundant Phylum (> 3%) in All Samples") +
  ylab("Relative Abundance") +
  theme(legend.position = "bottom") +
  scale_fill_manual(values = c('#1b9e77', '#d95f02', '#7570b3', '#a6761d', '#e7298a', '#666666', '#e6ab02', '#66a61e'))
p.comm.bar.t2

# Generate interavtive bar plot
ggplotly(p.comm.bar.t2)

# Time 3 (Day 7)
ps1_phylum.t3 <- ps1.t3 %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > threshold) %>%                    # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

# Convert Sample No to a factor because R is weird sometime
ps1_phylum.t3$patient_ID <- as.factor(ps1_phylum.t3$patient_ID)

# Creat a Phyla level table for external analysis
# write.table(ps1_phylum, file = "./results/phylum.txt", sep="\t")

# Plot - Phylum - Time 3 (Day 7)
p.comm.bar.t3 <- ggplot(ps1_phylum.t3, aes(x = patient_ID, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity") +
  facet_wrap(~randomization_arm, scales = "free_x") +
  ggtitle("Day 7\nAbundant Phylum (> 3%) in All Samples") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c('#1b9e77', '#d95f02', '#7570b3', '#a6761d', '#e7298a', '#e6ab02', '#66a61e')) +
  theme(legend.position = "bottom")
p.comm.bar.t3

# Color schemes
# Actionobacteria = #1b9e77
# Bacteroidetes = #d95f02
# Firmicutes = #7570b3
# Proteobacteria = #e7298a
# Verrumicrobia = #66a61e
# Synergistetes = #e6ab02
# Fusobacteria = #a6761d
# Spirochaetes = #666666

# Generate interavtive bar plot
ggplotly(p.comm.bar.t3)

grid.arrange(p.comm.bar.t1, p.comm.bar.t2, p.comm.bar.t3, nrow = 3)

```
Similar to the prevalence plots, there is a notable alteration in the Bacteroidetes and Firmicutes in antibiotic treated samples at Day 0. This is still noticeable at Day 7, but the Firmicutes are making somewhat of a comeback. Proteobacteria invade following antibiotic treatment as well, but are largely absent, particularly in the narrow spectrum antibiotic samples after 7 days. Overall, the restoration to the pre-antibiotic bacterial microbiome appears to be more prominent following narrow spectrum treatment than in broad spectrum treatment. 

##Alpha diversity summary information generation

```{r add-sample-data}
# Diversity
diversity <- global(ps1)
head(diversity)

sd.1 <- as.data.frame(sample_data(ps1)) # useful to coerce the phyloseq object into an R data frame
ps1.rich <- merge(sd.1, diversity, by ="row.names") # merge sd.1 by row names

# Add divergence measurements
ps1.rich$divergence <- divergence(ps1)

```
##Alpha diversity - categorical analysis

```{r alpha-diversity-day-treatment-richness}
# Subset by time (Day -9, 0 and 7)
ps1.rich.t1 <- subset(ps1.rich, day == "Pre")
ps1.rich.t2 <- subset(ps1.rich, day == "0")
ps1.rich.t3 <- subset(ps1.rich, day == "7")

my_comparisons <- list(c("No antibiotics", "Narrow spectrum antibiotics"), c("No antibiotics", "Broad spectrum antibiotics", c("Narrow spectrum antibiotics", "Broad specturm antibiotics")))

p.rich.t1.treat <- ggboxplot(ps1.rich.t1, x = "randomization_arm", y = "richness_0", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,260) +
  ylab("Richness") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means(comparisons = my_comparisons) +
  # stat_compare_means(label.y = 255) +
  ggtitle("Pre-antibiotics")

p.rich.t2.treat <- ggboxplot(ps1.rich.t2, x = "randomization_arm", y = "richness_0", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,260) +
  ylab("Richness") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means(comparisons = my_comparisons) +
  # stat_compare_means(label.y = 255) +
  ggtitle("Day 0")

p.rich.t3.treat <- ggboxplot(ps1.rich.t3, x = "randomization_arm", y = "richness_0", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,260) +
  ylab("Richness") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means(comparisons = my_comparisons) +
  # stat_compare_means(label.y = 255) +
  ggtitle("Day 7")

grid.arrange(p.rich.t1.treat, p.rich.t2.treat, p.rich.t3.treat, ncol = 3)

```

Richness (the number of taxa) is similar in all pretreatment sample groups. The number of taxa is greatly diminished in narrow and broad spectrum treatment samples compared to controls at day 0. This difference dissapears by day 7 when the number of bacterial taxa becomes similar between all treatment groups.

```{r alpha-diversity-day7-roto-boost-richness}
p.rich.t1.d7_roto_boost <- ggboxplot(ps1.rich.t1, x = "d7_roto_boost", y = "richness_0", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,260) +
  ylab("Richness") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means() +
  ggtitle("Pre-antibiotics")

p.rich.t2.d7_roto_boost <- ggboxplot(ps1.rich.t2, x = "d7_roto_boost", y = "richness_0", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,260) +
  ylab("Richness") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means() +
  ggtitle("Day 0")

p.rich.t3.d7_roto_boost <- ggboxplot(ps1.rich.t3, x = "d7_roto_boost", y = "richness_0", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,260) +
  ylab("Richness") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means() +
  ggtitle("Day 7")

grid.arrange(p.rich.t1.d7_roto_boost, p.rich.t2.d7_roto_boost, p.rich.t3.d7_roto_boost, ncol = 3)

```

```{r alpha-diversity-day7-roto-boost-shannon-div}
p.sd.t1.d7_roto_boost <- ggboxplot(ps1.rich.t1, x = "d7_roto_boost", y = "diversities_shannon", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,5) +
  ylab("Shannon Diversity") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means() +
  ggtitle("Pre-antibiotics")

p.sd.t2.d7_roto_boost <- ggboxplot(ps1.rich.t2, x = "d7_roto_boost", y = "diversities_shannon", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,5) +
  ylab("Shannon Diversity") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means() +
  ggtitle("Day 0")

p.sd.t3.d7_roto_boost <- ggboxplot(ps1.rich.t3, x = "d7_roto_boost", y = "diversities_shannon", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,5) +
  ylab("Shannon Diversity") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means() +
  ggtitle("Day 7")

grid.arrange(p.sd.t1.d7_roto_boost, p.sd.t2.d7_roto_boost, p.sd.t3.d7_roto_boost, ncol = 3)

```

```{r alpha-diversity-day-treatment-shannon_div}
p.rich.t1.treat.sd <- ggboxplot(ps1.rich.t1, x = "randomization_arm", y = "diversities_shannon", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,5) +
  ylab("Shannon Diversity") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means(comparisons = my_comparisons) +
  #stat_compare_means(label.y = 4.5) +
  ggtitle("Day -9")

p.rich.t2.treat.sd <- ggboxplot(ps1.rich.t2, x = "randomization_arm", y = "diversities_shannon", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,5) +
  ylab("Shannon Diversity") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means(comparisons = my_comparisons) +
  #stat_compare_means(label.y = 4.5) +
  ggtitle("Day 0")

p.rich.t3.treat.sd <- ggboxplot(ps1.rich.t3, x = "randomization_arm", y = "diversities_shannon", outlier.shape = NA) +
  geom_jitter(width = 0.2) +
  ylim(0,5) +
  ylab("Shannon Diversity") +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_compare_means(comparisons = my_comparisons) +
  #stat_compare_means(label.y = 4.5) +
  ggtitle("Day 7")

grid.arrange(p.rich.t1.treat.sd, p.rich.t2.treat.sd, p.rich.t3.treat.sd, ncol = 3)
# grid.arrange(p.rich.t1.treat, p.rich.t2.treat, p.rich.t3.treat, p.rich.t1.treat.sd, p.rich.t2.treat.sd, p.rich.t3.treat.sd, ncol = 3, nrow = 2)

```

Similar to richness, the diversity is similar in all pre-treatment samples. Diversity is decreased in both narrow and broad spectrum treated samples, but is restored by day 7. 

##ANCOVA analysis

```{r ANCOVA-BMI, include=FALSE, eval=FALSE}
# Test for age dependency on alpha diversity
# Age per time point
ancova(diversities_shannon~Age*day, data = ps1.rich)
ancova(richness_0~Age*day, data = ps1.rich)

# BMI
ancova(diversities_shannon~BMIcalc*day, data = ps1.rich)
ancova(richness_0~BMIcalc*day, data = ps1.rich)

# pre_RV_IgA
ps1.rich[, 91:94] <- log(ps1.rich[91:94], 2)
ancova(diversities_shannon~pre_RV_IgA*day, data = ps1.rich)
ancova(richness_0~pre_RV_IgA*day, data = ps1.rich)

# pneumo_IgA_pre
ancova(diversities_shannon~pneumo_IgA_pre*day, data = ps1.rich)
ancova(richness_0~pneumo_IgA_pre*day, data = ps1.rich)

# tetanus_IgG_pre
ancova(diversities_shannon~tetanus_IgG_pre*day, data = ps1.rich)
ancova(richness_0~tetanus_IgG_pre*day, data = ps1.rich)

```
Alpha diversity is not dependent on age, BMI, pre pneumo IgA, pre tetanus IgA or pre-rotavirus IgA for any of the three treatment groups. 

##Beta diversity analysis

```{r beta-diversity-arm}
# Ordination
ord.ps1.wuni <- ordinate(ps1.log, method = "PCoA", distance = "wunifrac")

# PCoA plot
p.ord.arm <- plot_ordination(ps1.log, ord.ps1.wuni, color = "randomization_arm") +
  geom_point(size=3, alpha = 0.7) +
  scale_color_brewer(palette = "Dark2") +
  geom_point(colour = "grey50", size = 1.5) +
  facet_grid(~ day) +
  stat_ellipse(geom = "polygon", alpha = 1/15, aes(fill = randomization_arm)) +
  labs(color = "randomization_arm")
p.ord.arm

```

```{r beta-diversity-d7-rv-boost}
# PCoA plot
p.ord.d7_boost <- plot_ordination(ps1.log, ord.ps1.wuni, color = "d7_roto_boost") +
  geom_point(size=3, alpha = 0.7) +
  scale_color_brewer(palette = "Dark2") +
  geom_point(colour = "grey50", size = 1.5) +
  facet_grid(~ day) +
  stat_ellipse(geom = "polygon", alpha = 1/15, aes(fill = d7_roto_boost)) +
  labs(color = "d7_roto_boost")
p.ord.d7_boost

```
##Premanova significance testing (ADONIS)

```{r ADONIS}
# Set a random seed so that exact results can be reproduced
set.seed(1000)

# Function to run adonis test on a physeq object and a variable from metadata 
doadonis <- function(physeq, category) {
  bdist <- phyloseq::distance(physeq, "bray")
  col <- as(sample_data(physeq), "data.frame")[ ,category]
  
  # Adonis test
  adonis.bdist <- adonis(bdist ~ col)
  print("Adonis results:")
  print(adonis.bdist)
  
  # Homogeneity of dispersion test
  betatax = betadisper(bdist,col)
  p = permutest(betatax)
  print("Betadisper results:")
  print(p$tab)
}

# Runs Permanov and Homogeneity of dispersion test
# See ?betadisper for more info
doadonis(ps1, "day")
doadonis(ps1.t1, "randomization_arm")
doadonis(ps1.t2, "randomization_arm")
doadonis(ps1.t3, "randomization_arm")

doadonis(ps1.con, "day")
doadonis(ps1.narrow, "day")
doadonis(ps1.broad, "day")

doadonis(ps1.t1, "d7_roto_boost")
doadonis(ps1.t2, "d7_roto_boost")
doadonis(ps1.t3, "d7_roto_boost")

```
##Random forest feature selection

```{r random-forest}
library(reshape2)
set.seed(2)
# Day 0 samples
output.forest.t2 <- randomForest(randomization_arm ~ pneumo_IgG_7 + d7_RV_IgA + d14_RV_IgA + d28_RV_IgA + pneumo_IgG_pre + pneumo_IgG_7 + pneumo_IgG_14 + pneumo_IgG_28 + tetanus_IgG_pre + tetanus_IgG_d7 + tetanus_IgG_d14 + tetanus_IgG_d28, na.action = na.omit, data = ps1.rich.t2)
print(output.forest.t2)

importance(output.forest.t2, type = 2)
importance.molten.t2 <- melt( importance(output.forest.t2, type = 2))
importance.molten.t2.sort <- arrange(importance.molten.t2, desc(value))
importance.molten.t2.sort$Var1 <- factor(importance.molten.t2.sort$Var1, levels = importance.molten.t2.sort$Var1)

p.rf.t2 <- ggplot(importance.molten.t2.sort, aes(x = Var1, y = value)) +
  geom_bar(stat = "identity") +
  ylab("Mean Decrease Gini (Importance)") +
  theme(axis.title.x = element_blank()) +
  #scale_x_discrete(labels=c("TNF-2", "Viral Load", "Shannon Diversity", "TNF-1", "sCD14", "Richness")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
  # annotate("text", x = 5, y = 2.5, label = "OOB error = 13.33%")
p.rf.t2

```

## Constrained correspondance analysis

```{r CCA}
# Calculate distance    
bray.cca <- phyloseq::distance(physeq = ps1.t1, method = "bray")

# CAP ordinate
cap_ord <- ordinate(
    physeq = ps1.t2, 
    method = "CAP",
    distance = bray.cca,
    formula = ~ tetanus_IgG_pre + tetanus_IgG_d7 + tetanus_IgG_d14 + tetanus_IgG_d28 + pneumo_IgG_pre + pneumo_IgG_7 + pneumo_IgG_14 + pneumo_IgG_28 + pre_RV_IgA + d7_RV_IgA + d14_RV_IgA + d28_RV_IgA
)

# CAP plot
cap_plot <- plot_ordination(
  physeq = ps1.t2, 
  ordination = cap_ord, 
    color = "randomization_arm", 
    axes = c(1,2)
) + 
    geom_point(aes(colour = randomization_arm), alpha = 0.4, size = 4) + 
    geom_point(colour = "grey90", size = 1.5)

# Now add the environmental variables as arrows
arrowmat <- vegan::scores(cap_ord, display = "bp")

# Add labels, make a data.frame
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)

# Define the arrow aesthetic mapping
arrow_map <- aes(xend = CAP1, 
    yend = CAP2, 
    x = 0, 
    y = 0, 
    shape = NULL, 
    color = NULL, 
    label = labels)

label_map <- aes(x = 1.3 * CAP1, 
    y = 1.3 * CAP2, 
    shape = NULL, 
    color = NULL, 
    label = labels)

arrowhead = arrow(length = unit(0.02, "npc"))

# Make a new graphic
cap_plot + 
  geom_segment(
    mapping = arrow_map, 
    size = .5, 
    data = arrowdf, 
    color = "gray", 
    arrow = arrowhead
  ) + 
  geom_text(
    mapping = label_map, 
    size = 4,  
    data = arrowdf, 
    show.legend = FALSE
  )

anova(cap_ord)

```
##Differential abundance testing - Randomization Arm - Pre vs. Day 0

```{r diffab-pre-vs-day0}
ps1.pre_d0 <- subset_samples(ps1, day != "7")

# Differential Abundance - Week
ds <- phyloseq_to_deseq2(ps1.pre_d0, ~randomization_arm)
dds <- DESeq(ds, test="Wald", fitType="parametric", betaPrior = FALSE) # Worth trying parametric and local fitTypes
res.dds.week = results(dds.week, cooksCutoff = FALSE)
alpha = 0.1 # Note: this is higher to try and allow even weak effect size taxa through
sigtab_dds.week = res.dds.week[which(res.dds.week$padj < alpha), ]
head(sigtab_dds.week)

write.table(sigtab_dds.week, file="./results/deseq_results_week.txt", sep = "\t")

# Quick check of factor levels
# Factor on the right will be on the left in the volcano plots. Relevel to switch
mcols(res.dds.week, use.names = TRUE)

# Create volcano plots of signficant results
# Prepare to join results data.table and taxonomy table
resdt.week = data.table(as(results(dds.week, cooksCutoff = FALSE), "data.frame"),
                   keep.rownames = TRUE)
setnames(resdt.week, "rn", "OTU")
taxdt.week = data.table(data.frame(as(tax_table(ps1), "matrix")), keep.rownames = TRUE)
setnames(taxdt.week, "rn", "OTU")

# Join results data.table and taxonomy table
setkeyv(taxdt.week, "OTU")
setkeyv(resdt.week, "OTU")
resdt.week <- taxdt.week[resdt.week]
resdt.week[, Significant := padj < alpha]
resdt.week[!is.na(Significant)]

volcano.week = ggplot(
  data = resdt.week[!is.na(Significant)][(pvalue < 1)],
  mapping = aes(x = log2FoldChange,
                y = -log10(pvalue),
                color = Phylum,
                label = OTU, label1 = Genus)) +
  theme_bw() +
  geom_point() + 
  geom_point(data = resdt.week[(Significant)], size = 7, alpha = 0.7) + 
  # geom_text(data = resdt.base[(Significant)], mapping = aes(label = paste("Genus:", Genus)), color = "black", size = 3) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) +
  geom_hline(yintercept = -log10(alpha)) +
  ggtitle("DESeq2 Negative Binomial Test Volcano Plot\nBaseline Samples") +
  theme(axis.title = element_text(size=12)) +
  theme(axis.text = element_text(size=12)) +
  theme(legend.text = element_text(size=12)) +
  geom_vline(xintercept = 0, lty = 2)
volcano.week

# Generate interactive plot
ggplotly(volcano.week)

```
