---
title: "Human Volunteer 16S Amplicon Preprocessing"
author: "Scott A. Handley & Barry Hykes, Jr"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: html_document
---
**Project Description:** Analysis of 16S rRNA amplicon data from a cohort of human volunteers (heatlhy men aged 18-35) randomized into three treatment groups: 1) control (no antibiotics), 2) narrow spectrum (oral vancomycin) or 3) broad-spectrum (oral vancomycin, ciprofloxacin, and metronidazole) antibiotics. Antibiotics were taken for 7-days prior to vaccination with Rotarix (RVV), polysaccharide-pneumococcal (Pneumo 23) and tetanus-toxoid vaccine. The primary endpoint was difference in 28 days-post-vaccination anti-RV IgA. Secondary endpoints were proportion of volunteers with day 7 anti-RV IgA boosting (>=2 fold-increase), absolute and proportion of RV-antigen shedding, anti-RV, pneumococcal and anti-tetanus IgG.

**Primary Collaborator:**
Vanessa Harris (v.harris@aighd.org)

**Notes:**
The following samples had 0 reads following demultiplexing: 20276.VanessaHarris, 20298.VanessaHarris, 20315.VanessaHarris, 20333.VanessaHarris, 20343.VanessaHarris, 20277.VanessaHarris, 20177.VanessaHarris, 20181.VanessaHarris, 20189.VanessaHarris, 20193.VanessaHarris, 20197.VanessaHarris, 20205.VanessaHarris, 20210.VanessaHarris, 20214.VanessaHarris, 20230.VanessaHarris, 20232.VanessaHarris, 20259.VanessaHarris

Directory locations will need to be updated to correspond to your local environment.

**Workflow details:** The R commands below represent a full analysis of the following:

1) Sequence quality filtering and trimming
2) Error rate inference
3) Chimera removal
4) Taxonomic assignment
5) Tree construction
6) Creation of phyloseq objects

**References:**
  ## http://f1000research.com/articles/5-1492/v1
  ## http://benjjneb.github.io/dada2/tutorial.html
  ## http://benjjneb.github.io/dada2/bigdata_paired.html

```{r initiate-environment}
# Set default knitr option
knitr::opts_chunk$set(fig.width=8,
                      fig.height=6,
                      fig.path="./figures/",
                      dev='pdf',
                      warning=FALSE,
                      message=FALSE)

# Load libraries
library("ShortRead"); packageVersion("ShortRead")
library("dada2"); packageVersion("dada2")
library("ggplot2"); packageVersion("ggplot2")
library("msa"); packageVersion("msa")
library("phangorn"); packageVersion("phangorn")
library("phyloseq"); packageVersion("phyloseq")

```
## Filter and trim the reads

Note: Settings need to be adjusted per individual sequencing run.
```{r filter-and-trim}
# File parsing
pathF <- "/mnt/data3/bhykes/16s/Harris/forward" # CHANGE ME to the directory containing your demultiplexed forward-read fastq files
filtpath <- file.path(pathF, "filtered/single") # Filtered forward files go into the pathF/filtered/ subdirectory
fastq <- sort(list.files(pathF, pattern="fastq"))

# Plot forward and reverse read quality
p.qual.f <- plotQualityProfile(pathF[1], aggregate = TRUE) + ggtitle("Fwd")
p.qual.f
filterAndTrim(fwd=file.path(pathF, fastq), filt=file.path(filtpath, fastq),
              trimLeft=10, truncLen=240, maxEE=2, truncQ=11, maxN=0, rm.phix=TRUE,
              compress=TRUE, verbose=TRUE, multithread=TRUE)

```
##Infer error rate

```{r sample-inference}
# Set paramaters
filtFs <- list.files(filtpath, pattern="fastq", full.names = TRUE)
sample.names <- sapply(strsplit(basename(filtFs), "_"), `[`, 1) # Assumes filename = samplename_XXX.fastq.gz
names(filtFs) <- sample.names

# Learn error rates
errF <- learnErrors(filtFs, nread=2e6, multithread=TRUE)
p.err.F <- plotErrors(errF, nominalQ = TRUE)
p.err.F

# Dereplicate, apply error rate to all sequences and merge
singles <- vector("list", length(sample.names))
names(singles) <- sample.names
for(sam in sample.names) {
  cat("Processing:", sam, "\n")
    derepF <- derepFastq(filtFs[[sam]])
    singles[[sam]] <- dada(derepF, err=errF, multithread=TRUE)
}
rm(derepF);

```
##Construct sequence table and remove chimeras

```{r seqtab-chimera-removal}
seqtab <- makeSequenceTable(singles)
seqtab.nochim <- removeBimeraDenovo(seqtab, multithread=TRUE)
saveRDS(seqtab.nochim, "seqtab.nochim.rds")

```
##Assign taxonomy

```{r assign-taxonomy}
# GreenGenes
taxa.gg <- assignTaxonomy(seqtab.nochim, "/mnt/data1/databases/dada2_taxonomy/gg_13_8_train_set_97.fa.gz", multithread = TRUE)
unname(head(taxa.gg))
colnames(taxa.gg) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# RDP
taxa.rdp <- assignTaxonomy(seqtab.nochim, "/mnt/data1/databases/dada2_taxonomy/rdp_train_set_16.fa.gz", multithread = TRUE)
unname(head(taxa.rdp))
colnames(taxa.rdp) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")

# Silva
taxa.silva <- assignTaxonomy(seqtab.nochim,  "/mnt/data1/databases/dada2_taxonomy/silva_nr_v128_train_set.fa.gz", multithread = TRUE)
unname(head(taxa.silva))
colnames(taxa.silva) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")

# HitDB
taxa.hitdb <- assignTaxonomy(seqtab.nochim, "/mnt/data1/databases/dada2_taxonomy/hitdb_v1.00.fa.gz", multithread = TRUE)

# Add species to RDP and Silva
#RDP
taxa.rdp.plus <- addSpecies(taxa.rdp, "/mnt/data1/databases/dada2_taxonomy/rdp_species_assignment_16.fa.gz")
unname(head(taxa.rdp.plus))
colnames(taxa.rdp.plus) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# SILVA
taxa.silva.plus <- addSpecies(taxa.silva, "/mnt/data1/databases/dada2_taxonomy/silva_species_assignment_v128.fa.gz")
unname(head(taxa.silva.plus))
colnames(taxa.silva.plus) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

save.image("/mnt/data3/bhykes/16s/Harris/Harris_M366_preprocessing_single.RData")
```
##Construct the phylogenetic tree

```{r construct-phylogeny}
seqs <- getSequences(seqtab.nochim)

names(seqs) <- seqs # This propagates to the tip labels of the tree
mult <- msa(seqs, method="ClustalW", type="dna", order="input")

phang.align <- as.phyDat(mult, type="DNA", names=getSequence(seqtab.nochim))

dm <- dist.ml(phang.align)
treeNJ <- NJ(dm) # Note, tip order != sequence order
fit = pml(treeNJ, data=phang.align)

fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
                       rearrangement = "stochastic", control = pml.control(trace = 0))
detach("package:phangorn", unload=TRUE)

```
##Create phyloseq objects

```{r create-phyloseq}
# Create PhyloSeq objects
# Greengenes
# Strip the r__ prefixes from taxon labels. This is only needed for the GreenGenes taxon annotations
taxa.gg.fixed <- gsub("k__", "", taxa.gg)
taxa.gg.fixed <- gsub("p__", "", taxa.gg.fixed)
taxa.gg.fixed <- gsub("c__", "", taxa.gg.fixed)
taxa.gg.fixed <- gsub("o__", "", taxa.gg.fixed)
taxa.gg.fixed <- gsub("f__", "", taxa.gg.fixed)
taxa.gg.fixed <- gsub("g__", "", taxa.gg.fixed)
taxa.gg.fixed <- gsub("s__", "", taxa.gg.fixed)

ps0.gg <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), tax_table(taxa.gg.fixed), phy_tree(fitGTR$tree))
ps0.gg

# RDP
ps0.rdp <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), tax_table(taxa.rdp.plus), phy_tree(fitGTR$tree))
ps0.rdp

# Silva
ps0.silva <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), tax_table(taxa.silva.plus), phy_tree(fitGTR$tree))
ps0.silva

# HitDb
ps0.hitdb <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), tax_table(taxa.hitdb), phy_tree(fitGTR$tree))
ps0.hitdb

# Sanity checks
get_taxa_unique(ps0.gg, "Phylum")
get_taxa_unique(ps0.rdp, "Phylum")
get_taxa_unique(ps0.silva, "Phylum")
get_taxa_unique(ps0.hitdb, "Phylum")

# Save RDS files for downstream analysis
saveRDS(ps0.gg, file = "/mnt/data3/bhykes/16s/Harris/ps0.Harris.gg_single.RDS")
saveRDS(ps0.rdp, file = "/mnt/data3/bhykes/16s/Harris/ps0.Harris.rdp_single.RDS")
saveRDS(ps0.silva, file = "/mnt/data3/bhykes/16s/Harris/ps0.Harris.silva_single.RDS")
saveRDS(ps0.hitdb, file = "/mnt/data3/bhykes/16s/Harris/ps0.Harris.hitdb_single.RDS")

save.image("/mnt/data3/bhykes/16s/Harris/Harris_M366_preprocessing_single.RData")

```
